pyproject.toml

```diff
@@ -34,10 +35,11 @@ numpy = "^1"
 pytz = "^2024.1"
 
 [tool.poetry.group.dev.dependencies]
-black = "^24.2.0"
+black = "24.2.0"
```

Comment:
Pinning the version of black as minor/micro version differences can be breaking. This guarantees that developers always format with the _exact_ same version of `black`, even if they've run `poetry update`.